deterministic_kernel.hpp

```diff
@@ -0,0 +1,289 @@
+    				// do not want anything put in the middle cell
+    				probability[i][j] = 0;
+    			} else {
+    				// store distance to center
+    				probability[i][j] = (std::abs(mid_row - i) / east_west_resolution)
+    								  + (std::abs(mid_col - j) / north_south_resolution);
+    				// determine probability based on distance
+    				if (kernel_type_ == DispersalKernelType::Cauchy) {
+ 					probability[i][j] = std::abs(cauchy.pdf(probability[i][j]));
+ 					} else if ( kernel_type_ == DispersalKernelType::Exponential ) {
+ 						probability[i][j] = std::abs(exponential.pdf(probability[i][j]));
+ 					}
+ 					sum += probability[i][j];
+ 				}
+    		}
+    	}
+    	//normalize based on the sum of all probabilities in the raster
+    	for ( int i = 0; i < row_max; i++ ) {
+    		for ( int j = 0; j < col_max; j++ ) {
+    			probability[i][j] /= sum;
+    		}
+    	}
+    }
+
+    /*! Generates a new position for the spread.
+     *
+     *  Creates smaller matrix "window" from probability matrix
+     *  New window created any time a new cell is selected from simulation.disperse
+     *
+     *  Selects next row/col value based on the cell with the highest probability
+     *  in the window.
+     *
+     *  Generator is not used
+     */
+    template<typename Generator>
+    std::tuple<int, int> operator() (Generator& generator, int row, int col)
+    {
+		// reset the window if considering a new cell
+    	if ( row != prev_row || col != prev_col ) {
+    		// determine dimensions for this window
+    		
+    		// determine the number of dispersers per cell and store these values
+    		// in the probability copy
+    		// dimensions from the original probability  matrix
+    		for ( int i = 0; i < row_max; i++) {
+    			for ( int j = 0; j < col_max; j++) {
+    				probability_copy[i][j] = probability[i][j];
+    			}
+    		}
+    	}
+    	
+    	int row_translate = row - mid_row;
+    	int col_translate = col - mid_col;
+    	
+    	double max = (double)-std::numeric_limits<int>::max();
+    	// have to add a max value because if row == probability_r_min and the probability
+    	// of this cell is greater than every other cell - no cell will be selected
+    	int max_prob_row = 0;
+    	int max_prob_col = 0;
+    	
+    	//find cell with highest probability
+    	for ( int i = 0; i < row_max; i++ ) {
+    		for ( int j = 0; j < col_max; j++ ) {
+    			// compare row/col to middle row/col
+    			if ((i + row_translate)== row && ((j + col_translate) == col)) {
+    				// do nothing
+    				// should be zero anyways - could probably remove this
+    			}
+    			else if (probability_copy[i][j] > max ) {
+    				max = probability_copy[i][j];
+    				max_prob_row = i;
+					max_prob_col = j;
+    			}
+    		}
+    	}
+
+		// subtracting 1 preserves the probability order but ensures that this cell
+		// is not select again unless all other cells in window have been selected
+    	probability_copy[max_prob_row][max_prob_col] -= 1;
```

Comment:
It should subtract 1- (1/number_of_dispersers). The 1/number of dispersers should be initialized with the probability matrix based on the value in the disper matrix on the first pass (i.e. if row and column are new get the value from the dispersers matrix otherwise 1/ number of dispersers should be used for the rest of the dispersers from that cell to modify the probability. Doing it this way ensures we always move the same proportion of the individuals to each cell no matter how many are dispersing.